### PR TITLE
Rename all references of import to be plugin

### PR DESCRIPTION
**Pull request by cam-stitt**
In preparation of some changes that are happening to the Sentinel runtime, we are renaming import to plugin in all locations.

This is a backwards incompatible change and will require a minor bump of the sdk.

The failing tests are expected due to the cyclic dependency between the sdk and sentinel.

### CODE REVIEW

**Comment by glennsarti** [comment]
> This is a backwards incompatible change

If this is a breaking change then shouldn't this be a major bump?

---

Does this require consumers of the SDK to recompile their plugins? Feels like this is missing some migration steps for consumers.

Will Sentinel support both types of RPC services? `.Import` and `.Plugin`?

**Comment by cam-stitt** [comment]
> > This is a backwards incompatible change
> 
> If this is a breaking change then shouldn't this be a major bump?
> 
> Does this require consumers of the SDK to recompile their plugins? Feels like this is missing some migration steps for consumers.
> 
> Will Sentinel support both types of RPC services? `.Import` and `.Plugin`?

`sentinel-sdk` currently has no major version. We are currently not yet at v1.0, so this would bump the minor version the same way prior breaking changes have.

File: framework/interface.go [hunk 1]
```diff
@@ -1,25 +1,23 @@
 package framework
 
 //go:generate rm -f mock_*.go
-//go:generate mockery -inpkg -note "Generated code. DO NOT MODIFY." -name=Root -testonly
-//go:generate mockery -inpkg -note "Generated code. DO NOT MODIFY." -name=Namespace -testonly
+//go:generate mockery --inpackage --note "Generated code. DO NOT MODIFY." --name=Root --testonly
+//go:generate mockery --inpackage --note "Generated code. DO NOT MODIFY." --name=Namespace --testonly
+// go:generate mockery --inpackage --note "Generated code. DO NOT MODIFY." --name=NamespaceCreator --testonly
```
Comment (mrinalirao) [inline framework/interface.go:6]: extra space between the // and the go:generate

File: framework/interface.go [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-
-// Root is the import root. For any import, there is only a single root.
-// For example, if you're implementing an import named "time", then the "time"
-// identifier itself represents the import root.
+// Root is the plugin root. For any plugin, there is only a single root.
+// For example, if you're implementing a plugin named "time", then the "time"
+// identifier itself represents the plugin root.
 //
-// The root of an import is configurable and is able to return the actual
+// The root of an plugin is configurable and is able to return the actual
```
Comment (mrinalirao) [inline framework/interface.go:12]: nit: The root of `a` plugin

File: proto/plugin.proto [hunk 3]
```diff
@@ -27,7 +27,7 @@ message Configure {
     }
 }
 
-// Get are the structures for an Import.Get.
+// Get are the structures for an Plugin.Get.
```
Comment (mrinalirao) [inline proto/plugin.proto:30]: nit: a Plugin.Get.

**Review by mrinalirao — APPROVED** [review]

**Pull request merged**